### PR TITLE
fix for  flux plugin: combine two separate code paths for populating the cache in resync() scenario and regular case #4117 

### PR DESCRIPTION
**Pull request by gfichtenholt**
Worked on this for a bit and this is as elegant I as can make it thus far. I have other things I need to attend to.

### CODE REVIEW

**Comment by gfichtenholt** [comment]
Antonio/Michael, pls review this. Ideally, it would be Dimitri reviewing this, since he's the one who asked for this change, but I am not sure whether he's attached to the project at the moment. This change basically removes a special case where I bypass the workqueue during a cache resync operation, so that even during resync all the work is done via the queue. Also a bunch of DRYing up in unit test cases. Thanks in advance. Greg

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/test_util_test.go [hunk 1]
```diff
@@ -234,6 +238,59 @@ func installedRef(id, namespace string) *corev1.InstalledPackageReference {
 	}
 }
 
+func newCtrlClient(repos []sourcev1.HelmRepository, charts []sourcev1.HelmChart, releases []helmv2.HelmRelease) withWatchWrapper {
+	// register the flux GitOps Toolkit schema definitions
+	scheme := runtime.NewScheme()
+	_ = sourcev1.AddToScheme(scheme)
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/test_util_test.go:244]: Not sure why you're assigning to `_` here? Isn't that necessary only if you want some returned values but not others?

**Review by absoludity — APPROVED** [review]
Excellent Greg.

**Comment by gfichtenholt** [comment]
> Excellent Greg.

Thank you Michael

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/watcher_cache.go [hunk 2]
```diff
@@ -39,6 +40,9 @@ const (
 	// max number of attempts to resync before giving up
 	maxWatcherCacheResyncBackoff = 2
 	KeySegmentsSeparator         = ":"
+	// max number of concurrent workers computing or retrieving retrieving cache
```
[the remark this answers is not in the document]
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/watcher_cache.go:43]: will fix

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/test_util_test.go (hunk 1 above)
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/test_util_test.go:244]: just followed flux's example code:
https://fluxcd.io/docs/gitops-toolkit/packages/#crud-example

**Pull request merged**